SelectRequestInput.jsx

```diff
@@ -1,59 +1,57 @@
-    // the hash in the link will be used to immediately display the specific data.
-    useEffect(() => {
-        if (window.location.hash) {
-          const hash_value = window.location.hash.split("#")[1];
-          setLinkHashValue(hash_value);
-        }
-    }, [window.location.hash])
-    
-    useEffect(() => {
-      if (link_hash_value.length > 0) {
-        const find_select_value = playground_requests.find(el => el.name === link_hash_value);
-        setHashValueTitle(find_select_value?.title);
-      }
-    }, [link_hash_value, playground_requests])
-
-    // This useEffect has mainly a UX purpose, so that the user can see
-    // what function they selected in the link (hash).
-    useEffect(() => {
-      if (selected_value && request_body) {
-        window.location.hash = request_body ? request_body.name : window.location.hash || "";
-      }
-    }, [selected_value, request_body, window.location.hash])
+const SelectRequestInput = ({ selected, setSelected, handleChange }) => {
+  const [isActive,setIsActive] = useState(false)
+  const [toggle, setToggle] = useState(false)
+  library.add(faCaretDown);
+  library.add(faCaretUp);
+  const [searchResults, setSearchResults] = useState("");
 
     return (
-      <fieldset className={styles.apiRequest}>
-        <select
-          className={`${styles.select2} ${selected_value != "Select API Call - Version 3" ? styles.selected : ""}`}
-          onChange={handleChange}
-          defaultValue={link_hash_value ? hash_value_title : default_value}
-          id="settings-dropdown"
-        >
-          <option disabled className={styles.option}>{link_hash_value ? hash_value_title : default_value}</option>
-          <optgroup label="All calls">
-            {playground_requests.map(el => (
-              <option value={el.name} key={el.name}>
-                {el.title}
-              </option>
-            ))}
-          </optgroup>
-        </select>
-      </fieldset>   
+    <fieldset>
+      <div className={styles.dropdown}>
+      <div className= {styles.dropdownBtn} onClick={() => {
+        setIsActive(!isActive); 
+        setToggle(!toggle)
+      }}>
+        {selected}
+        {!isActive ? <FontAwesomeIcon icon="fa-solid fa-caret-down" /> : <FontAwesomeIcon icon="fa-solid fa-caret-up" /> }
+      </div>
+      {isActive && (
+        <div className= {`${styles.dropdownContent} ${toggle ? styles.show : ''}`}>
+          <input type="text" id="myInput" className= {styles.dropdownSearch} onChange={event =>setSearchResults(event.target.value)}/>
+          <div className={styles.dropdownItem}> Select API Call - Version 3</div>
+          <div className={styles.dropdownStart}>ALL CALLS</div>
+          {playground_requests.filter(option =>{
+            if(searchResults === ""){
+              return option;
+            } else if ( option.title.toLowerCase().includes(searchResults.toLowerCase())){
+              return option;
+            }
```

Comment:
- Both cases are returning `option`! 
- Condition needs to be simplified
- `Else` condition is missing which is not a good practice